### PR TITLE
Delete XmonGate class

### PR DESCRIPTION
**Pull request by Strilanc**
Part of https://github.com/quantumlib/Cirq/issues/952

### CODE REVIEW

**Comment by Strilanc** [comment]
Just a few more little cuts...

File: cirq/google/programs.py [hunk 1]
```diff
@@ -68,8 +68,8 @@ def _x_to_proto_dict(gate: ops.XPowGate, q: ops.QubitId) -> Dict:
     exp_w = {
         'target': cast(devices.GridQubit, q).to_proto_dict(),
         'axis_half_turns':
-            xmon_gates.XmonGate.parameterized_value_to_proto_dict(0),
-        'half_turns': xmon_gates.XmonGate.parameterized_value_to_proto_dict(
+            xmon_gates._parameterized_value_to_proto_dict(0),
```
Comment (dabacon) [inline cirq/google/programs.py]: always slightly strange when a module calls another modules _private method.  Maybe it doesn't belong in cirq.google but might still make it non _private.

File: cirq/google/xmon_gates.py [hunk 2]
```diff
@@ -27,117 +26,96 @@
...
+        raise ValueError('invalid operation: {}'.format(proto_dict))
+
+
+def _parameterized_value_from_proto_dict(message: Dict
+                                         ) -> Union[value.Symbol, float]:
+    if 'raw' in message:
+        return message['raw']
+    if 'parameter_key' in message:
+        return value.Symbol(message['parameter_key'])
+    raise ValueError('No value specified for parameterized float.')
```
Comment (dabacon) [inline cirq/google/xmon_gates.py]: show message in error string.

File: cirq/google/programs.py (hunk 1 above)
Comment (Strilanc) [reply cirq/google/programs.py]: Yup. There's an upcoming change where the xmon gates file is disappearing and will merge into programs.py.

File: cirq/google/xmon_gates.py (hunk 2 above)
Comment (Strilanc) [reply cirq/google/xmon_gates.py]: done.

**Comment by Strilanc** [comment]
All covered now

File: cirq/google/xmon_gates.py [hunk 3]
```diff
@@ -14,9 +14,8 @@
 
 """Gates that can be directly described to the API, without decomposition."""
 
-from typing import Dict, Optional, Union, Tuple, Any, cast
+from typing import Dict, Union, Tuple, Any, cast
```
Comment (dabacon) [inline cirq/google/xmon_gates.py]: nit: alphabetical

Comment (Strilanc) [reply cirq/google/xmon_gates.py]: Acknowledged. Leaving as is because this file is not long for this world.

**Pull request merged**